### ISSUE TITLE
build failed on linux - `System.ArgumentNullException: Value cannot be null. (Parameter 'element')`

### ISSUE THREAD

**Issue by whoracle** [hugoattal/ElectronicNodes#93]
**Describe the bug**
Building the plugin on linux fails.

**To Reproduce**
1. Buy the Plugin ;)
2. Download the Plugin on Windows
3. Copy the Folder to your Linux install
4. Remove `Binaries` and `Intermediate` folders
5. Run the following (substitute your paths, obv.):
```
export UE_ENGINE_PATH="/opt/unreal-engine/Engine" # or wherever your UE installation lives
export PLUGIN_PATH="/home/username/tmp/ElectronicNodes/ElectronicNodes.uplugin"
export PACKAGE_PATH="/home/username/git/games/mygame/Plugins/ElectronicNodes"

${UE_ENGINE_PATH}/Build/BatchFiles/RunUAT.sh BuildPlugin -plugin=${PLUGIN_PATH} -package=${PACKAGE_PATH} -Rocket -TargetPlatforms=Linux -PackageDir=${PACKAGE_PATH}
```
6. Observe the build error in `/home/username/Library/Logs/Unreal Engine/LocalBuildLogs/UBA-UnrealEditor-Linux-Development.txt` - See attached [build log](https://github.com/user-attachments/files/17701153/UBA-UnrealEditor-Linux-Development.txt).

**Expected behavior**
Plugin should be built successfully

**Environment (please complete the following information):**
 - Unreal Engine version: 5.4.4
 - Operating System: Linux (ArchLinux)

**Additional Information**
- I recently built PaperZD from source, so there's nothing *fundamentally* wrong with my system/UE installation. Might of course be missing something more specific for ElectronicNodes, though
- The error message is, IMHO, supremely unhelpful, else I'd have tried to fix it myself. It doesn't point to a code error in one of the plugin files or something, but still - One plugin was built correctly, the other wasn't, so here I am.

**Comment by hugoattal** [comment]
Hey there, sorry for that...
It's really weird and, as you mentioned, the error message is pretty unhelpful 😅.

Can you try removing the `-Rocket -TargetPlatforms=Linux` flags?

Or use the automatic compilation of the editor here https://github.com/hugoattal/ElectronicNodes/wiki/How-to-install-Electronic-Nodes-on-previous-version-of-the-engine%3F-On-custom-build%3F ?

**Comment by whoracle** [comment]
> Can you try removing the -Rocket -TargetPlatforms=Linux flags?

Just did - no change, same error.

Will try the autocompilation later, however something similar failed for me with PaperZD a few days back. But we'll see, maybe I'll be lucky today :)

**Comment by whoracle** [comment]
As a quick test, I used the 5.4.0 version sources that I tried to compile earlier into /opt/unreal-engine. On Startup UE complained about the Plugin not being built for 5.4 (of course, since I changed the version in the uplugin file to 5.3 and it is missing binaries). However, no prompt for recompilation shows up. I can try to enable it in Plugins, but after a restart nothing changed. Seems Epic doesn't support startup recompilation on Linux, or at least it's disabled in some way in the defaults...

Will have a deeper look tomorrow.

**Comment by hugoattal** [comment]
That's super weird, seems like it's an old issue from 5.3:

https://forums.unrealengine.com/t/exception-while-generating-include-data-for-unrealeditor-system-argumentnullexception-value-cannot-be-null-parameter-element/1332875

https://forums.unrealengine.com/t/ubt-bug-no-longer-able-to-compile-plugins-in-standalone-since-5-3/1318659

Could you try this: https://forums.unrealengine.com/t/ubt-bug-no-longer-able-to-compile-plugins-in-standalone-since-5-3/1318659/11 ?

**Comment by whoracle** [comment]
OK, back with some more info:

> Could you try this

Since I don't have VisualStudio, I am unable to recompile that. I can do it in my Windows install, but then I'd get a .dll which won't help me on my Linux box.

I did recompile PaperZD in the meantime, to see if I had somehow broken my local install - that one went through without a hitch. Documentation on compiling UE stuff on Linux, let alone troubleshooting, is few and far between, and nothing I found so far has helped me.

I will try compiling the plugin for windows using my fresh 5.4.4 install and the built-in recompilation tonight, just to make sure it's _actually_ something related to my box/setup, and not something that Epic borked somewhere along the line where the plugin won't build at all on current UE.

My full version string for UE is `5.4.0-33043543+++UE5+Release-5.4`, as reported by trying to load/recompile the plugin on startup (why that does not say 5.4.4-whatever is beyond me... the UE boot splash claims 5.4.4.)

**Comment by whoracle** [comment]
Update: I reinstalled UE (same zip as before, don't ask me what I screwed up along the way, I don't know), and afterwards Plugin compilation worked without a hitch.

I'll keep the issue open if you have any further questions, and I can always try to recompile and reproduce it with the borked installation which I saved in a different folder.

Otherwise, feel free to close this and then I'll delete my borked install copy.
